packages/fxa-auth-server/lib/error.js

```diff
@@ -1422,6 +1424,19 @@ AppError.iapInvalidToken = (error) => {
   );
 };
 
+AppError.iapPurchaseConflict = (error) => {
+  const extra = error ? [{}, undefined, error] : [];
+  return new AppError(
+    {
+      code: 422,
```

Comment:
I thought 422 was reserved for WebDAV. I think this is still closest to a conflict, or maybe a 403 as they're attempting a 'forbidden action'.

Reply:
TIL: [WebDAV](https://en.wikipedia.org/wiki/WebDAV). I don't see any mention of WebDAV in the [422 MDN docs](https://developer.mozilla.org/en-US/docs/web/http/status/422). We already use this code elsewhere in our error library ([`AppError.cannotSendEmail`](https://github.com/mozilla/fxa/blob/ee71759ff2c282a876c0e7e0ddb3a4e888aefc0d/packages/fxa-auth-server/lib/error.js#L995-L1000)). I've also seen it [used by Zendesk in Sentry](https://sentry.io/organizations/mozilla/issues/3302142606/?environment=prod&query=is%3Aunresolved+unprocessable+entity&statsPeriod=14d). Jeremy on the VPN team also suggested this code.

After looking at the MDN docs for both codes, 422 seems better, as it indicates retrying the request will not work and unlike 403, is not associated with authorization. That said, I'm open to another response code if there is one that's more apt.

Reply:
Okay thanks for the extra context; I've updated the response code to be a `403` Forbidden.